packages/eslint-plugin/docs/rules/no-unnecessary-type-assertion.mdx

```diff
@@ -73,6 +73,16 @@ function foo(x: number | undefined): number {
 
 ## Options
 
+### `ignoreStringConstAssertion`
```

Comment:
[Feature] The root issue also happens with other literals `bigint`, `boolean`, `number`. This PR will have to fix those cases too. How about naming this `ignoreLiteralConstAssertion`

Example case of `123`/`number`: https://typescript-eslint.io/play/#ts=5.4.2&fileType=.tsx&code=KYDwDg9gTgLgBAYwgOwM7wLYEMoGthRwC8cMUArsANwCwAUPaJLHAJbIwEBmWCwcAESwwscAN704cbHgIAuUgE8wwCF2k58UWnSkA3LABtKC9FHYBzHQF96jcNHhdyyBDFYpEUYMOBCRABQAlAr%2BohK6iCjocABGWKj8JGIaslAKMlpw1jpS3jDkUMjicAB05fGJADRwBsbACgBEjdk2QA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0ipWkOTJE0fJQ5N0UOdA7RI4MAF8QOoA&tsconfig=N4KAvkA&tokens=false

Reply:
I was forced to fix the boolean case as it wasn't working but otherwise I think I've taken review into account! Thanks